sink/sink

```diff
@@ -412,17 +416,76 @@ def mkdir_and_chdir(base, identifier):
     os.fchmod(dirfd, 0o755)
     os.close(dirfd)
 
+def prune_logs(base_path=".", dry_run=False):
+    """ Check all subdirectories of the current path and delete the ones that
+        are either empty or contain only files older than the age threshold.
+
+        For dry runs, print the name of the directory instead.
+    """
+    reference_time = time.time()
+    age_threshold = 30
+    entries = os.listdir(base_path)
+    for entry in entries:
+        path = os.path.join(base_path, entry)
+
+        # initialize to threshold so empty dirs will be deleted
+        last_modified = 0
+        # use the last modified time of the logfile itself, not just the directory
+        for root, dirs, files in os.walk(path):
+            for filename in files:
+                last_modified = max(last_modified, os.stat(os.path.join(root, filename)).st_mtime)
+
+        diff_days = (reference_time - last_modified) / (60*60*24)
+
+        # keep anything newer than 30 days old
+        if diff_days < age_threshold:
+            continue
+
+        if dry_run:
+            sys.stderr.write("Pruning {0} (age: {1} days)\n".format(path, int(round(diff_days))))
+        else:
+            shutil.rmtree(path)
+
+def check_prune(config, logs, dry_run):
+    """ See if we have to prune logs, and if we should, fork and do so
+    """
+    prune_interval = config.get("Sink", "PruneInterval")
```

Comment:
prune_interval is a string here, and Python will happily compare strings to numbers... so `int(config.get("Sink", "PruneInterval"))`?